front-end-notes/angular.md

```diff
@@ -0,0 +1,112 @@
+
+```js
+angular.module('app' ,[]);
+ 
+.service('movie', function () {
+  this.title = 'The Matrix';
+});
+ 
+.controller('MyController', function (movie) {
+  expect(movie.title).toEqual('The Matrix');
+});
+```
+
+####Factory
+
+A factory is a service that can be injected as a function. A factory is a lot like a service in the sense that it is a singleton and dependencies can be specified in the function. The difference between a factory and a service is that a factory injects a plain function so AngularJS will call the function and a service injects a constructor. A constructor creates a new object so new is called on a service and with a factory you can let the function return anything you want. As you will see later on, a factory is a provider with only a $get method.
+
+```js
+angular.module('app', []);
+ 
+.factory('movie', function () {
+  return {
+    title: 'The Matrix';
+  }
+});
+ 
+.controller('MyController', function (movie) {
+  expect(movie.title).toEqual('The Matrix');
+});
+```
+
+```js
+.factory('catalogueService', function($rootScope, $http) {
+  // We first define a private API for our service.
+
+  // Private vars.
+  var items = [];
+
+  // Private methods.
+  function add( id ) {
+    $http.put( $rootScope.apiURL, {id:id} )
+    .success(function(data,status,headers,config) { items.push(data); })
+    .then(function(response) { console.log(response.data); });
+  }
+
+  function store( obj ) {
+    // do stuff
+  }
+
+  function remove( obj ) {
+    // do stuff
+  }
+
+  // We now return a public API for our service.
+  return {
+    add: add,
+    store: store,
+    rm: remove
+  };
+};
+```
+
+### Filters
+
+A filter formats the value of an expression for display to the user. They can be used in view templates, controllers or services and it is easy to define your own filter.
+#####example:
+
+```html
+<input type="text" ng-model="searchText" />
+```
+
+Now we add the `filter` condition to our `ng-repeat` helper. Since we used `ng-model` above to create a two-way binding on the `searchText` variable, Angular will automatically match your search string against **any** key in the Todo 
+
+```html
+<div>
+  <ul ng-repeat="todo in todos | filter: searchText">
+    <li>{{todo.name}}</li>
+  </ul>
+</div>
+```
+
+### Partials
+
+Partials are a way to link the view (html) to the model and controller using two-way data-binding.
+####Two-way data-binding just means that:
+1. When properties in the model get updated, so does the UI.
+2. When UI elements get updated, the changes get propagated back to the model.
+
+##Sources
+https://docs.angularjs.org/api?PHPSESSID=cae8e98e7ca559b4605d75c813b358ee
+http://www.tomdalling.com/blog/software-design/model-view-controller-explained/

```

Comment:
Two of the links were on the same line. You may want to break them up for easier readability. Suggestion would be using * for listing items in markdown.